notebooks/official/workbench/spark/spark_sample_notebook.ipynb

```diff
@@ -0,0 +1,1219 @@
+{
```

Comment:
<p>Based on the schema printed above, data of the GitHub Activity is not stored in primitive types, but is instead stored in arrays.  </p><p>To work more effectively with the data, you need to <strong>convert </strong><s>preprocess</s> it to primitive types and separate data for monoglot repos and polyglot repos.<s> Once you create preprocessed columns, it makes our future tasks much faster. </s></p><p>You can see three Python functions with the <code>@udf</code> annotation with their return type below. The annotation <code>@udf</code> is a short form of <a href="https://spark.apache.org/docs/3.1.3/api/python/reference/api/pyspark.sql.functions.udf.html" target="_blank">User Defined Function</a>, which is used to extend the functions of the PySpark framework. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/720/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>